perf-regression/polaris/run_gpu_margo_p2p_bw.qsub

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+#PBS -l select=2:system=polaris
+#PBS -l place=scatter
+#PBS -l walltime=0:10:00
+#PBS -l filesystems=home
+#PBS -q debug
+#PBS -A radix-io
+
+# Set to location of installed mochi tests directory
+# where the gpu-margo-p2p-bw program should be available
+# in the 'bin' directory
+MOCHI_TESTS_INSTALL=/home/jhendersonhdf/mochi-tests-build
+
+set -eu
+
+# Change to working directory
+cd ${PBS_O_WORKDIR}
+
+# note: disable registration cache for verbs provider for now; see
+#       discussion in https://github.com/ofiwg/libfabric/issues/5244
+export FI_MR_CACHE_MAX_COUNT=0
+# use shared recv context in RXM; should improve scalability
+export FI_OFI_RXM_USE_SRX=1
+
+echo "Setting up Spack"
+source $HOME/spack/share/spack/setup-env.sh
+echo "Activating env"
+spack env activate mochi-env
+spack find -fN
+
+# Just in case..
+spack load mochi-margo@develop ^libfabric+cuda
+
+echo "Running Margo P2P BW GPU Test"
+
+echo "=== Testing transfer from GPU memory of Host A to GPU memory of Host B ==="
+mpiexec -n 2 -ppn 1 $MOCHI_TESTS_INSTALL/bin/gpu-margo-p2p-bw -x 4096 -n "ofi+verbs://" -c 1 -D 10
```

Comment:
I suspect this is failing with the "unable to cudaMalloc..." error because it is asking for too big of a total memory buffer on the device; perhaps it does not allow a single allocation that large.  You can specify this on the command line with the -X option.  It must be a multiple of the value of the -x option.  The -X specifies total memory buffer size on each host, while -x is the transfer size.  It will loop over the buffer doing rdma operations of size x until time runs out.  If -x and -X are set to the same value then it repeatedly transfers the same buffer.

Reply:
The program was failing with CUDA error 222, so there was some mismatch between the NVCC that the program gets compiled with vs. the CUDA version that it runs with. Note that I was using the GNU programming environment on Polaris so things might get a little tricky there. Originally I was using the `cudatoolkit-standalone` module, which was using NVCC version 11.8 for compiling and running, but this seems to be problematic. After loading the `nvhpc-mixed` module instead, according to https://www.alcf.anl.gov/support/user-guides/polaris/compiling-and-linking/compiling-and-linking-overview/index.html, I can run the program successfully. I also adjusted the script to use a 500MiB total buffer size with 1MiB transfers just to be safe.

So it seems like the script runs correctly, but I'm still not sure how to describe the correct way to compile it. I probably need to start from a fresh environment and detail that.

Reply:
Great, glad you found a working scenario.  I'm going to leave this PR open for now to see if we can resolve the compiler settings.